modules/thermal_hydraulics/doc/content/source/components/HSBoundaryExternalAppHeatFlux.md

```diff
@@ -0,0 +1,56 @@
+# HSBoundaryExternalAppHeatFlux
+
+This component is a [heat structure boundary](thermal_hydraulics/component_groups/heat_structure_boundary.md)
+that is used to apply a heat flux variable transferred from another application
+in a Neumann boundary condition.
+Currently, this component may only be used with 2D, cylindrical heat structures,
+such as [HeatStructureCylindrical.md].
+
+## Usage
+
+This component creates a heat flux variable with the name given by
+[!param](/Components/HSBoundaryExternalAppHeatFlux/heat_flux_name). The other
+application is responsible for transferring into this variable, and this name
+will need to be supplied to the other application's input file. If
+[!param](/Components/HSBoundaryExternalAppHeatFlux/heat_flux_is_monomial) is set
+to `true`, then the finite element type of the variable will be `CONSTANT MONOMIAL`;
+otherwise, it will be `FIRST LAGRANGE`. The former is recommended whenever the
+other application uses a layered-average user object like [NearestPointLayeredSideAverageFunctor.md],
+since in each axial division, there is only one value; there is not a value for
+each *node*. The latter is recommended whenever some kind of nodal transfer is
+used.
+
+The parameter [!param](/Components/HSBoundaryExternalAppHeatFlux/heat_flux_is_inward)
+indicates whether the transferred heat flux corresponds to the *inward* direction,
+with respect to the heat structure boundary, i.e., if `true`, a *positive* value
+indicates heat is being transferred *to* the heat structure, rather than *from* it.
```

Comment:
you dont support extracting the heat flux from the HS boundary so I would not say "from it".

I would talk about the sign simply

Reply:
I reworded, so you can check. I guess the confusion was the word "transfer". I meant the physical process, not the MOOSE thing.